jscomp/ml/typecore.ml

```diff
@@ -1273,7 +1273,7 @@ and type_pat_aux ~constrs ~labels ~no_existentials ~mode ~explode ~env
         | _            -> k None
       end
   | Ppat_record(lid_sp_list, closed) ->
-      assert (lid_sp_list <> []);
+      (*assert (lid_sp_list <> []);*)
```

Comment:
Had a look at where this is triggered: it's when calling `Parmatch.ppat_of_type` during pattern match exhaustiveness check.
The check turns typed pattern match into an untyped one (in this case with 0 fields) and then performs type checking on the result.

But directly, it would be triggered by this:
```res
| B({}) => Js.log("B")
```

I think we should explore allowing this, by just removing this from `bs_ast_invariant.ml`:
```ml
| Ppat_record ([], _) ->
          Location.raise_errorf ~loc:pat.ppat_loc
            "Empty record pattern is not supported"
```

and check that it also works for non-nested records.